src/main/java/frc/robot/commands/AutoMotionProfiling.java

```diff
@@ -0,0 +1,125 @@
+package frc.robot.commands;
+
+import frc.robot.subsystems.drivetrain.DriveBase;
+import frc.robot.Constants;
+import java.util.List;
+import edu.wpi.first.math.controller.RamseteController;
+import edu.wpi.first.math.controller.SimpleMotorFeedforward;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.kinematics.ChassisSpeeds;
+import edu.wpi.first.math.kinematics.DifferentialDriveKinematics;
+import edu.wpi.first.math.kinematics.DifferentialDriveWheelSpeeds;
+import edu.wpi.first.math.trajectory.Trajectory;
+import edu.wpi.first.math.trajectory.TrajectoryConfig;
+import edu.wpi.first.math.trajectory.TrajectoryGenerator;
+import edu.wpi.first.math.trajectory.Trajectory.State;
+import edu.wpi.first.math.trajectory.TrajectoryParameterizer.TrajectoryGenerationException;
+import edu.wpi.first.math.trajectory.constraint.CentripetalAccelerationConstraint;
+import edu.wpi.first.math.trajectory.constraint.DifferentialDriveVoltageConstraint;
+import edu.wpi.first.math.trajectory.constraint.TrajectoryConstraint;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+
+public class AutoMotionProfiling extends CommandBase {
+    private static final double ramseteB = 2;
+    private static final double ramseteZeta = 0.7;
+
+    private final DriveBase drive;
+    private final DifferentialDriveKinematics kinematics;
+    private final Trajectory trajectory;
+    private final RamseteController controller =
+        new RamseteController(ramseteB, ramseteZeta);
+    private final Timer timer = new Timer();
+
+    public AutoMotionProfiling(DriveBase drive, double startVelocity,
+        List<Pose2d> waypoints, double endVelocity, boolean reversed) {
+        this(drive, startVelocity, waypoints, endVelocity, reversed, List.of());
+    }
+
+    public AutoMotionProfiling(DriveBase drive, double startVelocity,
+        List<Pose2d> waypoints, double endVelocity, boolean reversed,
+        List<TrajectoryConstraint> constraints) {
+        addRequirements(drive);
+        this.drive = drive;
+
+        double maxVoltage, maxVelocity,
+            maxAcceleration, maxCentripetalAcceleration;
+        switch(Constants.bot) {
+            case ROBOT_2022_COMP:
+            case ROBOT_2022_PRACTICE:
+                maxVoltage = 10.0;
+                maxVelocity = 2.5;
+                maxAcceleration = 20.0;
```

Comment:
This is max 2.5 meters per second! That's really fast
```suggestion
                maxVelocity = 0.5;
                maxAcceleration = 1;
```